sunspec/core/util.py

```diff
@@ -95,6 +96,12 @@ def data_to_double(data):
         return d[0]
 
 def data_to_str(data):
+
+    # Change the data from bytes string to regular string for python 3
+    # compatibility
+    if sys.version_info > (3,):
+        data = str(data, 'latin-1')
```

Comment:
In regards to specifying in the spec, that's Bob's domain, but it might not be necessary. The reason I introduced encoding and decoding in the first place was just trying to switch back and forth between byte string to regular string and had nothing to do with formatting. I just needed the exact same string as the other type because certain functions required certain forms. `latin-1` just happened to be the format that I found provided this functionality.

ex. `"\xc2\xa5"` -> `b"\xc2\xa5"` and vice versa

Looking through your code from when you made the library python 3 compatible I believe you solved this issue with `from builtins import str`. I'm going to look more into this approach in the time I have this coming week. I think we got too into the weeds of implementation and I lost track of what the code was trying to accomplish in the first place.